vdr/vdr.go

```diff
@@ -162,7 +162,17 @@ func (r VDR) Update(id did.DID, current hash.SHA256Hash, next did.Document, _ *t
 		return err
 	}
 
-	key, err := r.keyStore.Resolve(controllers[0].CapabilityInvocation[0].ID.String())
+	var key crypto.Key
+outer:
+	for _, c := range controllers {
+		for _, cik := range c.CapabilityInvocation {
+			key, err = r.keyStore.Resolve(cik.ID.String())
+			if err == nil {
+				break outer
+			}
+		}
+	}
+
 	if err != nil {
 		if errors.Is(err, crypto.ErrKeyNotFound) {
 			return types.ErrDIDNotManagedByThisNode
```

Comment:
The error on line 180 is now incorrect because it insinuates a single controller, but all controllers are checked.

`return fmt.Errorf("could not find capabilityInvocation key for controller: %w", err)`

This should be something along the lines of "could not find capabilityInvocation key for updating the DID document"